app/assets/javascripts/discourse/app/lib/wait-for-keyboard.js

```diff
@@ -0,0 +1,66 @@
+import { getOwner } from "@ember/application";
+
+export async function waitForKeyboard(context) {
+  return new Promise((resolve) => {
+    const safeguard = setTimeout(() => {
+      // eslint-disable-next-line no-console
+      console.warn("Keyboard visibility didnt change after 1s.");
+
+      resolve({
+        visible:
+          document.documentElement.classList.contains("keyboard-visible"),
+      });
+    }, 1000);
+
+    if (!window.visualViewport) {
+      return resolve({ visible: false });
+    }
```

Comment:
I reckon we should move the timeout after all of the 'immediate' return checks. We only need the safeguard once we reach the `onViewportResize` step?

Reply:
done